scripts/pre_commit_linter.py

```diff
@@ -505,6 +541,27 @@ def _check_newline_character(all_files):
     return summary_messages
 
 
+def _check_bad_pattern_in_file(filename, content, pattern):
+    """
+        Detects whether the given pattern is present in the file.
+        Args:
+            filename: str. Name of the file.
+            content: str. Contents of the file.
+            pattern: dict. Object containing details for the pattern
+                to be checked.
+    """
+    regexp_pattern = pattern['regexp']
+    if not (any(filename.startswith(bad_pattern)
+                for bad_pattern in pattern['excluded_dirs'])
```

Comment:
Why is the dummy variable named bad_pattern? Shouldn't it be something like excluded_dir (given that you're already using the term "pattern" for other things)?

Reply:
oh right. Renamed to excluded_dir.